cpp/ql/src/semmle/code/cpp/ir/implementation/aliased_ssa/Instruction.qll

```diff
@@ -40,7 +40,7 @@ module InstructionSanity {
         opcode instanceof Opcode::Chi and tag instanceof ChiTotalOperandTag or
         opcode instanceof Opcode::Chi and tag instanceof ChiPartialOperandTag or
         (
-          (opcode instanceof ReadSideEffectOpcode or opcode instanceof MayWriteSideEffectOpcode) and
+          (opcode instanceof ReadSideEffectOpcode or opcode instanceof MayWriteSideEffectOpcode or opcode instanceof Opcode::InlineAsm) and
```

Comment:
I think you should just make `InlineAsm` extend `MayWriteSideEffectOpcode`.

Reply:
Ugh, that made the IR dump look like the `InlineAsm` instruction is a side effect of the previous instruction (i.e. `^InlineAsm`). Please either fix that, or just revert to what you had before, whichever is easier.